### ISSUE TITLE
Certain colormaps do not work with Sodium

### ISSUE THREAD

**Issue by razorchase49** [MehVahdJukaar/polytone#116]
Minecraft Version: 1.21.1
Fabric Version: 0.16.9
Polytone Version: 1.21-2.5.12

I play with PureBDcraft and many other resourcepacks which use OptiFine's vanilla colormap format (a 256x256 .png), which is what Polytone is meant to be compatible with. And it is, unless I am using Sodium. I know Sodium version 0.5 brought changes that had to make modders and resourcepack authors rewrite the way to implement custom colors, but seeing how packs like Golden Days and the sample Polytone pack are able to use colormaps that work even when using up-to-date Sodium, I thought the process would be automatic for every other resourcepack. But custom colors for things like stone-like blocks and sand do not change color if Sodium is present. Although things like sky and fog color still change. And, if you are using a minimap like Xaero's, you can see the color changes on the minimap. I've been trying to draw lines as to what make Golden Days and the sample Polytone pack work, and been trying to match other resourcepacks to follow the same format, but no luck.
To replicate:
Load Minecraft with Fabric, using no mods besides Polytone and Sodium, and any resourcepack that uses the OptiFine's vanilla colormaps (John Smith Legacy, PureBDcraft, Ozocraft, etc.). Go to either a Desert or a Badlands biome, and you'll see that, although the sky changes color, the sand / red sand remains unchanged as though you are not using custom colors.
![2024-11-06_14 02 23](https://github.com/user-attachments/assets/aa6d4b03-8f9a-4d98-aba7-55704c752f1d)
Now, simply remove Sodium, and you'll see that the sand / red sand properly changes color.
![2024-11-27_11 07 37](https://github.com/user-attachments/assets/74522746-f14c-4a80-9c18-9205c56361e6)
I'm wondering what the solution is. Currently thinking if there's a way to convert the vanilla colormaps to grid format, so they may _potentially_ make it past Sodium. Still trying to figure out why Golden Days and the sample Polytone pack work just fine.

**Comment by MehVahdJukaar** [comment]
First of all polytone compat layer is an after thought and hacky in many places. By far the last of the mod concerns as stuff is meant to be written using poltrone data format. 
As for sodium if you say stuff works without it's 100% their issue. Sodium has been known to hardcodes vanilla systems in the past, stuff like water colours for example or not being able to register custoon biome colour sampers. Still most of these are solved (finally) in the latest sodium update which is 0.6. are you sure you are using that?

**Comment by razorchase49** [comment]
Yes, I'm using Sodium 0.6. I can imagine it's 100% their issue; the problem is, they just don't update very actively at all like you do, so I thought reporting here would lead to a more immediate resolution.
For now, do you know if there's some sort of tool to convert a 256x colormap into a format that would work even with Sodium, much like Golden Days and your sample pack?

**Comment by MehVahdJukaar** [comment]
You would need to go into the specifics of which colormsp is not working. Try doing that in polytone format. 
Idk what you mean by converting, poltrone colormsp textures can be any size
